### ISSUE TITLE
Project status?

### ISSUE THREAD

**Issue by 09croberts** [projectdelphai/panorama-tab-groups#19]
Sorry if this is rude, but what's the status of this project?

At least a month ago this released with good pr that it was going to be a maintained and continually improved version of panorama tab groups. If there hasn't been a commit in a month then it would seem it has fallen fate to the same issues of not being updated. We want good tab groups, and this project initially showed great promise, but it kinda has to keep being updated.

Best

**Comment by erakli** [comment]
+1

Also, why this repo was created? There's already one fork of original repo https://github.com/projectdelphai/panorama-view.

**Comment by projectdelphai** [comment]
Not rude at all don't worry. Basically it boils down to:

 1. I received a message [here](https://github.com/photodiode/panorama-view/issues/8#issuecomment-420119975) that it was just in hibernation and that he would be coming back in mid-november. I'm slightly hesitant to create drastic changes in the code considering that at any point they might come back and it will be really hard to merge everything back in. I'm not opposed to big changes (and I have them going in secondary branches on my private repo) but that leads to the second point which is:
 1. I'm not good enough at javascript/webextensions/understand photodiode's repo well enough to understand how to make super big changes. Bug fixes or small changes are easy enough (such as the keyboard shortcuts) but things such as moving pinned tabs to the status bar on the right or auto-tiling grids are much harder. For example, I spent a week or so after work figuring out the code to move pinned tabs to the status bar (using another fork) to find out that it only sometimes worked/bugged out and I had to scrap the code cause it was working. Similarly, I tried to autoclose empty tab groups after all the tabs inside were deleted and it only worked about 60% of the time. And while it might be fine for personal use, considering I'm releasing this for other people to use, I'm not comfortable releasing features that aren't 100% working. It just boils down to I'm the only person working on this and I can't churn out features as fast as people might like considering the remaining features are big ones (in my opinion at least).
 1. Lastly, I won't deny that for me, the biggest issue personally was being able to drag a new tab onto empty space to create a new group. After that was done, I had some other priorities first so this project fell down a little bit. I still managed to merge pull requests if they come in (i've only gotten one from other contributors) and small fixes such as keyboard shortcuts or messaging support, but progress has waned solely since I'm the only one working on it. 

As for why this repo was created instead of the fork, it's mostly cause I wanted to upload the changes to the Mozilla add-on website so that others could use my changes, but I didn't want to mislead users into thinking that this was the same panorama view. I just wanted to keep the repos solely different (in case photodiode never came back) and creating a new project was easier than keeping a fork going. I deleted the other fork to keep things simpler. I wasn't sure if that was the right approach to take but no one messaged me or complained so I just went with it.

**Comment by projectdelphai** [comment]
tl;dr I'm here but I'm slow so features will come out when I have time after work and only when I'm happy that it's bug free and usable by other people.

**Comment by erakli** [comment]
Sounds reasonable. 

I'm totally agree with 100%-idealogy, but I think, there's need in small maintenance in issues. I talked about at least marking them with proper tags. Furthemore, it would be more reliable, if you point out which features are in progress by you and what are still on todo. 

I cannot help in first place with repo because of total abscence of knowledge in JS, but I could take some time with small features. I need understanding about things that are 

1. could be taken by JS-newbie
2. doesn't need much time time to spent

**Comment by projectdelphai** [comment]
went through most of the issues and tagged them, also anything i have is already pushed to the code. Sorry about any stagnation you might have seen but especially with the influx of contributors, I think progress is starting up again. hopefully you should see some more changes coming. I'll close this issue for now unless there's any other questions you have :)
